lua/keymappings.lua

```diff
@@ -1,86 +1,108 @@
--- better window movement
-vim.api.nvim_set_keymap("n", "<C-h>", "<C-w>h", { silent = true })
-vim.api.nvim_set_keymap("n", "<C-j>", "<C-w>j", { silent = true })
-vim.api.nvim_set_keymap("n", "<C-k>", "<C-w>k", { silent = true })
-vim.api.nvim_set_keymap("n", "<C-l>", "<C-w>l", { silent = true })
+local function register_mappings(mappings, default_options)
+  for mode, mode_mappings in pairs(mappings) do
+    for _, mapping in pairs(mode_mappings) do
+      local options = #mapping == 3 and table.remove(mapping) or default_options
+      local prefix, cmd = unpack(mapping)
+      pcall(vim.api.nvim_set_keymap, mode, prefix, cmd, options)
+    end
+  end
+end
 
--- TODO fix this
--- Terminal window navigation
-vim.cmd [[
-  tnoremap <C-h> <C-\><C-N><C-w>h
-  tnoremap <C-j> <C-\><C-N><C-w>j
-  tnoremap <C-k> <C-\><C-N><C-w>k
-  tnoremap <C-l> <C-\><C-N><C-w>l
-  inoremap <C-h> <C-\><C-N><C-w>h
-  inoremap <C-j> <C-\><C-N><C-w>j
-  inoremap <C-k> <C-\><C-N><C-w>k
-  inoremap <C-l> <C-\><C-N><C-w>l
-  tnoremap <Esc> <C-\><C-n>
-]]
+local mappings = {
+  n = {
+    -- better window movement
+    { "<C-h>", "<C-w>h", { silent = true } },
+    { "<C-j>", "<C-w>j", { silent = true } },
+    { "<C-k>", "<C-w>k", { silent = true } },
+    { "<C-l>", "<C-w>l", { silent = true } },
 
--- TODO fix this
--- resize with arrows
-if vim.fn.has("mac") == 1 then
```

Comment:
You removed the if that control the mappings for macs.

Reply:
It is line 87

Reply:
sorry, i missed that.